src/test/shared/utilities/timeoutUtils.test.ts

```diff
@@ -0,0 +1,54 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as assert from 'assert'
+import * as timeoutUtils from '../../../shared/utilities/timeoutUtils'
+
+describe ('timeoutUtils', async () => {
+
+    describe ('Timeout', async () => {
+
+        it ('returns > 0 if the timer is still active', async () => {
+            const longTimer = new timeoutUtils.Timeout(300)
+            assert.strictEqual(longTimer.remainingTime > 0, true)
+            // bleed the timeout to not mess with other tests
+            await longTimer.timer
```

Comment:
These kinds of tests compound to cause the whole test suite to run longer. Is it possible to avoid waiting, perhaps by introducing a method onto `Timeout` that would cause the promise to immediately end? The test would then run in (likely) < 50 ms instead of 300 ms.

Reply:
✔️